snapcraft/internal/elf.py

```diff
@@ -147,7 +148,7 @@ def get_elf_files(root: str, file_list: Sequence[str]) -> FrozenSet[str]:
     if ms.load() != 0:
         raise RuntimeError('Cannot load magic header detection')
 
-    elf_files = set()
+    elf_files = dict()  # type: Dict[str, Dict[str, str]]
```

Comment:
As this functionality grows, I wonder if it would make more sense to make the `Dict[str, str]` part of this a class instead.

Reply:
the last `str` of that is probably going to morph into an `Any`. I don't mind making it a class, and given your comment below I might as well.